wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Redactor/Autosave.js

```diff
@@ -2,75 +2,57 @@
-            createOverlay: function () { },
-            hideOverlay: function () { },
-            _saveToStorage: function () { },
-            _cleanup: function () { }
-        };
-        return Fake;
-    }
+    Core = tslib_1.__importStar(Core);
+    Devtools_1 = tslib_1.__importDefault(Devtools_1);
+    EventHandler = tslib_1.__importStar(EventHandler);
+    Language = tslib_1.__importStar(Language);
+    UiRedactorMetacode = tslib_1.__importStar(UiRedactorMetacode);
     // time between save requests in seconds
-    var _frequency = 15;
-    /**
-     * @param       {Element}       element         textarea element
-     * @constructor
-     */
-    function UiRedactorAutosave(element) { this.init(element); }
-    UiRedactorAutosave.prototype = {
+    const _frequency = 15;
+    class UiRedactorAutosave {
         /**
          * Initializes the autosave handler and removes outdated messages from storage.
          *
          * @param       {Element}       element         textarea element
          */
-        init: function (element) {
+        constructor(element) {
             this._container = null;
-            this._metaData = {};
             this._editor = null;
-            this._element = element;
             this._isActive = true;
             this._isPending = false;
-            this._key = Core.getStoragePrefix() + elData(this._element, 'autosave');
-            this._lastMessage = '';
-            this._originalMessage = '';
-            this._overlay = null;
+            this._lastMessage = "";
+            this._metaData = {};
+            this._originalMessage = "";
             this._restored = false;
             this._timer = null;
+            this._element = element;
+            this._key = Core.getStoragePrefix() + this._element.dataset.autosave;
+            //this._overlay = null;
             this._cleanup();
             // remove attribute to prevent Redactor's built-in autosave to kick in
-            this._element.removeAttribute('data-autosave');
-            var form = DomTraverse.parentByTag(this._element, 'FORM');
+            delete this._element.dataset.autosave;
+            const form = this._element.closest("form");
             if (form !== null) {
-                form.addEventListener('submit', this.destroy.bind(this));
+                form.addEventListener("submit", this.destroy.bind(this));
             }
             // export meta data
-            EventHandler.add('com.woltlab.wcf.redactor2', 'getMetaData_' + this._element.id, (function (data) {
-                for (var key in this._metaData) {
-                    if (this._metaData.hasOwnProperty(key)) {
-                        data[key] = this._metaData[key];
-                    }
-                }
-            }).bind(this));
+            EventHandler.add("com.woltlab.wcf.redactor2", `getMetaData_${this._element.id}`, (data) => {
+                Object.entries(this._metaData).forEach(([key, value]) => {
```

Comment:
I believe this could be: `data = { ...data, ...this._metaData }`. I'll leave it up to you whether you like this from a clarity perspective.

Reply:
I'm not too happy about it, it's shorter but it feels like "clever code" to me. It also reminds me of `Core.extend()` which implies that there are default values in `data`.